commit d4fdd1a78821884cf515467cdbbb769096033452
Author: Tom Harrison <l12436@yahoo.com.tw>
Date:   Wed Aug 20 21:10:53 2014 +0200

    Fix custom folder list in slideshow wallpaper config
    
    REVIEW: 118180
    BUG: 327580
    FIXED-IN: 4.11.12

diff --git a/libs/plasmagenericshell/backgrounddialog.cpp b/libs/plasmagenericshell/backgrounddialog.cpp
index 645de3f..4f93e57 100644
--- a/libs/plasmagenericshell/backgrounddialog.cpp
+++ b/libs/plasmagenericshell/backgrounddialog.cpp
@@ -480,8 +480,10 @@ void BackgroundDialog::changeBackgroundMode(int index)
     }
 
     if (d->wallpaper) {
-        d->wallpaper->setPreviewing(true);
-        d->preview->setPreview(d->wallpaper);
+        if (mode == "SingleImage") {
+            d->wallpaper->setPreviewing(true);
+            d->preview->setPreview(d->wallpaper);
+        }
         d->wallpaper->setRenderingMode(mode);
         KConfigGroup cfg = wallpaperConfig(plugin);
         //kDebug() << "making a" << plugin << "in mode" << mode;
